### PR TITLE
Issue #1 Japanese locale file for XUI

### PR DESCRIPTION
**Pull request by ogis-miyamura**
## Analysis
In order to enable Japanese display with XUI, add Japanese locale file for XUI.

## Solution
Add the following four Japanese locale files.

```
openam-ui / openam-ui-ria / src / main / resources / locales
├─ja
│      authorize.json
│      translation.json
│
└─ja-JP
        authorize.json
        translation.json
```

## Install/Update
N/A

## Compatibility
Please refer to item "I18N".

## Performance
N/A

## I18N
When you set the language setting of the web browser to "ja" or "ja-JP", the message displayed in XUI will be in Japanese.

## Testing

- WEB browser used for testing
  - Chrome
  - Firefox
  - IE
  - Edge

- Language settings
  - en
  - ja
  - ja-JP

## Regression testing
N/A

### CODE REVIEW

**Comment by tsujiguchitky** [comment]
Please also test with IE or Edge. In these browsers, `ja-JP` is specified in the Accept-Language header. `ja` is used in Chrome and Forefox.

**Comment by ogis-miyamura** [comment]
Thank you for review.
Even when using IE and Edge, XUI language has been changed by language setting change.
So added Tested items.

**Comment by tsujiguchitky** [comment]
Considering the maintenance, it is redundant to create the country-specific locale file.
How about changing the behavior of i18nManager.js(i18next.js)?

``` 
index.html
 21         <script type="text/javascript">
 22             var require = {
 23                 urlArgs : "v=${version}",
 24                 deps : ['main'],
 25                 config : {
 26                     'org/forgerock/commons/ui/common/main/i18nManager': {
 27                         i18nLoad: 'none'
 28                     }
 29                 }
 30             };
 31         </script>
```

In this case, it works even without `ja-JP`.

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 1]
```diff
@@ -0,0 +1,1292 @@
...
+                "changesSaved": "変更を保存しました。"
+            },
+            "CommonMessages" : {
+                "authenticationFailed" : "ユーザー名/パスワードの組み合わせが不正です。",
+                "serviceUnavailable" : "サービスが利用できません。",
+                "unknown" : "不明なエラー。システム管理者にお問い合わせください。",
+                "loggedIn" : "正常にログインしました。",
+                "errorFetchingData" : "ユーザーデータフェッチ中のエラー。",
+                "loggedOut" : "ログアウトしました。",
+                "loginTimeout" : "ログインセッションがタイムアウトになりました。再起動しています...",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `再起動しています...` -> `最初からやり直してください。`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 2]
```diff
@@ -0,0 +1,1292 @@
...
+            }
+        }
+    },
+    "templates" : {
+        "user" : {
+            "LoginDialog" : {
+                "refreshOnLogin" : "ログイン時にリフレッシュする必要があります" 
+            },
+            "LoginTemplate" : {
+                "logintextconter" : "",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: What is `logintextconter`? It is not in the original file.

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 3]
```diff
@@ -0,0 +1,1292 @@
...
+                "loginRemember" : "ユーザー名を記憶する。",
+                "noAccountQuestion" : "アカウントを持っていないですか？",
+                "registerAccount" : "アカウントを登録してください。",
+                "problemLoggingQuestion" : "ログインのトラブルがありますか？",
+                "socialAuthentication" : "または次でログイン:",
+                "resetPassword" : "パスワードをリセットしますか？",
+                "forgotUsername" : "ユーザー名を忘れましたか?",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: Please unify character size of question mark(Full-width or half-width).

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 4]
```diff
@@ -0,0 +1,1292 @@
...
+            "propertyName": "プロパティ名",
+            "propertyValue": "プロパティ値",
+            "addName": "名の追加",
+            "addValue": "値の追加",
+            "validation" : {
+                "errorsNotSaved" : "フォームに検証エラーが存在します。変更は保存できません。",
+                "VALID_PHONE_FORMAT": "無効な文字が含まれています。",
+                "VALID_NAME_FORMAT": "無効な文字が含まれています。",
+                "VALID_EMAIL_ADDRESS_FORMAT": "電子メールアドレスが無効です。",
+                "AT_LEAST_X_CAPITAL_LETTERS" : "最低 __numCaps__ capital 文字以上は必要です。",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `～文字以上の大文字が必要です。`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 5]
```diff
@@ -0,0 +1,1292 @@
...
+                "AT_LEAST_X_NUMBERS" : "少なくとも __numNums__ 個以上の数字が必要です。",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: Please unify the words(`少なくとも` or `最低`).

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 6]
```diff
@@ -0,0 +1,1292 @@
...
+                "CANNOT_CONTAIN_OTHERS" : "値を含めることはできません: __disallowedFields__",
+                "MIN_LENGTH" : "最低 __minLength__ 文字です。",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `～文字必要です。`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 7]
```diff
@@ -0,0 +1,1292 @@
...
+                "REQUIRED" : "空白にすることはできません。",
+                "UNIQUE" : "すでに存在します。",
+                "REAUTH_REQUIRED" : "",
+                "formContainsErrors" : "フォームには検証エラーが含まれています。",
+                "atLeastOneCapitalLetter" : "少なくとも大文字が 1 文字以上必要です。",
+                "atLeastOneNumber" : "少なくとも数字が 1 つ以上必要です。",
+                "atLeast8Characters" : "少なくとも 8 文字以上必要です。",
+                "cannotMatchLogin" : "ログインが一致しません。",
+                "confirmationMatchesPassword" : "新規パスワードとパスワード(確認)が一致。",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `～一致する必要があります。`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 8]
```diff
@@ -0,0 +1,1292 @@
...
+                "usernameExists" : "ユーザー名がすでに存在します。",
+                "emailAddressAlreadyExists" : "電子メールアドレスがすでに存在します。",
+                "onlyAlphabeticCharacters" : "英字のみです。",
+                "onlyNumbersAndSpecialCharacters" : "数字と特殊文字のみです。",
+                "cannotMatchOldPassword" : "古いパスワードと一致しません。",
+                "minimum4Characters" : "最小 4 文字です。",
+                "acceptanceRequiredForRegistration" : "登録には受理が必要です。",
+                "incorrectPassword" : "パスワードが正しくありません。",
+                "incorrectSecurityAnswer" : "応答が無効です。",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `回答が正しくありません。`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 9]
```diff
@@ -0,0 +1,1292 @@
...
+            "common" : {
+                "confirmType": "サイト"
+            },
+            "edit": {
+                "title" : "サイト設定"
+            },
+            "list" :{
+                "title" : "サイト",
+                "new" : "サイトの追加",
+                "table" : ["サイト名", "URL", "セカンダリURL", "割り当てられたサーバー"],
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: Please insert a space before and after English words.

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 10]
```diff
@@ -0,0 +1,1292 @@
...
+            "quickStart" : "共通タスク"
+        },
+        "services": {
+            "edit": {
+                "title": "サービス",
+                "titleNew": "サービスの追加",
+                "chooseServiceType": "サービスタイプを選択",
+                "serviceTypes": "サービスタイプ...",
+                "settings": "設定",
+                "unavaliable": "加えることが可能なサービスではありません。",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `追加できるサービスはありません。`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 11]
```diff
@@ -0,0 +1,1292 @@
...
+                "activateMessage": "これはアクティブなレルムです。削除した場合は、すべての設定データが失われます。設定データを保持してログインを防ぎたい場合は、レルムを非アクティブに設定することができます。<br>このレルムを削除しますか?",
+                "deactivateMessage": "これは、非アクティブなレルムです。現在誰もログインすることはできません。削除した場合は、すべての設定データが失われます。<br>このレルムを削除してもよろしいですか?"
+            },
+            "realmStatusPopover": {
+                "title": "レルムの状態",
+                "content": "レルムがアクティブではない場合、すべての OpenAM サービスはこのレルム内のすべてのユーザーに対して使用できなくなります。"
+            },
+            "realmAliasesPopover": {
+                "title": "レルム / DNS エイリアス",
+                "content": "認証ユーザーインターフェースがリクエストが受信すると、ユーザーが認証されるべきレルムを見つけるため、 OpenAM はすべてのレルムに対してこの属性を検索します。 OpenAM は一致するレルムを1つだけを見つける必要しかないので、複数のレルムに重複したエントリを入れてはいけません。さらに OpenAM が一致するレルムを見つけられない場合、ユーザーにはエラー画面が表示されます。"
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `OpenAM は一致するレルムを1つに特定する必要があるため、～`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 12]
```diff
@@ -0,0 +1,1292 @@
...
+                "title": "新規スクリプト",
+                "language": "言語",
+                "script": "スクリプト",
+                "scriptType": "スクリプトタイプ",
+                "validation": {
+                    "line": "行",
+                    "column": "列",
+                    "message": "メッセージ",
+                    "errors": "このスクリプトにはエラーがあります:",
+                    "noErrors": "エラーは見つかりませんでした."
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `.` -> `。`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 13]
```diff
@@ -0,0 +1,1292 @@
...
+                },
+                "dialog": {
+                    "title":  "スクリプトタイムの変更",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `スクリプトタイプ`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 14]
```diff
@@ -0,0 +1,1292 @@
...
+                "grid": ["名前", "スクリプトタイプ", "言語", "説明"],
+                "confirmDeleteText": "選択したスクリプトを削除してもよろしいですか?"
+            }
+        },
+        "configuration" : {
+            "authentication" : {
+                "description": "グローバルな認証機能のデフォルトプロパティを設定します。",
+                "core": {
+                    "coreAttributes": "コア属性",
+                    "title": "デフォルト認証レルム",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `レルムのデフォルト設定`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 15]
```diff
@@ -0,0 +1,1292 @@
...
+                    "description": "新しいレルムを作成するときに適用するデフォルトの認証プロパティを設定します。"
+                },
+                "modules": {
+                    "title": "デフォルト認証モジュール",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `認証モジュールのデフォルト設定`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 16]
```diff
@@ -0,0 +1,1292 @@
...
+                "noActions": "アクションがありません...",
+                "noActionsSelected": "選択されたアクションがありません...",
+                "noResources": "選択したリソースがありません...",
+                "noResourceTypes": "選択されたリソースタイプがありません...",
+                "noStaticAttributes": "静的な属性がありません...",
+                "noSubjectAttributes": "サブジェクト属性がありません...",
+                "noApplications": "選択したポリシーセットがありません...",
+                "noPatterns": "パターンがありません...",
+                "noResourceTypesAvailable": "利用可能なリソースタイプがありません...",
+                "noApplicationsAvailable": "使用可能なポリシーセットがありません...",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: Please unify the words(`使用可能` or`利用可能`).

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 17]
```diff
@@ -0,0 +1,1292 @@
...
+                    "description": "このポリシーセットに適用される認可ポリシーを作成、編集、削除します。",
+                    "noPolicies": "このポリシーセットにポリシーを追加していません。"
+                }
+            },
+
+            "resourceTypes": {
+                "list": {
+                    "add": "新規リソースタイプ",
+                    "title": "リソースタイプ",
+                    "description": "適用されるポリシーリソースとそれらのリソースで実行できるアクション用のテンプレートを定義します。",
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: `ポリシーが適用されるリソースと～`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/authorize.json [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+            "build": "Build"
+        }
+    },
+    "form": {
+        "description": "このアプリケーションは、以下の個人情報を要求しています:",
+        "noScopes": "このアプリケーションは、あなたのアカウントへのアクセスを要求しています",
+        "signedInAs": "ログインユーザー:",
+        "save": "同意状態を保存",
+        "allow": "許可",
+        "deny": "不許可"
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/authorize.json]: `不許可` -> `拒否`

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 1 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I think that it is a reasonable proposal, thank you.
```suggestion
                "loginTimeout" : "ログインセッションがタイムアウトになりました。最初からやり直してください。",
```

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 2 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: The line is a typo, so delete that line.
```suggestion
            "LoginTemplate" : {
                "loginRemember" : "ユーザー名を記憶する。",
```

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 4 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I think that it is a reasonable proposal, thank you.
```suggestion
                "AT_LEAST_X_CAPITAL_LETTERS" : "最低 __numCaps__ capital 文字以上の大文字が必要です。",
```

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 6 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I think that it is a reasonable proposal, thank you.
I would like to make the following contents as familiar with the items above.
```suggestion
                "MIN_LENGTH" : "最低 __minLength__ 文字以上は必要です。",
```

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 7 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I think that it is a reasonable proposal, thank you.
```suggestion
                "confirmationMatchesPassword" : "新規パスワードとパスワード(確認)が一致する必要があります。",
```

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 8 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I think that it is a reasonable proposal, thank you.
```suggestion
                "incorrectSecurityAnswer" : "回答が正しくありません。",
```

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 15 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I think that it is a reasonable proposal, thank you.
```suggestion
                    "title": "認証モジュールのデフォルト設定",
```

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/authorize.json (hunk 18 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/authorize.json]: It is typo.
```suggestion
        "deny": "拒否"
```

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 17 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I think that it is a reasonable proposal, thank you.
```suggestion
                    "description": "ポリシーが適用されるリソースと～とそれらのリソースで実行できるアクション用のテンプレートを定義します。",
```

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 5 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: There were other similar problems.
However, because each group is unified, I will change only the place pointed out.
```suggestion
                "AT_LEAST_X_NUMBERS" : "最低 __numNums__ 個以上の数字が必要です。",
```

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 16 above)
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: There were many other similar problems.
I will unify expressions "使用".

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 19]
```diff
@@ -0,0 +1,1292 @@
...
+                "resources": "リソース",
+                "subjects": "対象",
+                "environments": "環境",
+                "resourceType": "リソースタイプ",
+                "resourceTypes": "リソースタイプ",
+                "defaultState": "デフォルトの状態",
+
+                "noRealms": "選択したレルムがありません...",
+                "noActions": "アクションがありません...",
+                "noActionsSelected": "選択されたアクションがありません...",
```
Comment (ogis-miyamura) [inline openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I will unify the expression '選択した'.
There were many other similar problems.

**Comment by ogis-miyamura** [comment]
> Considering the maintenance, it is redundant to create the country-specific locale file.
> How about changing the behavior of i18nManager.js(i18next.js)?
> 
> ```
> index.html
>  21         <script type="text/javascript">
>  22             var require = {
>  23                 urlArgs : "v=${version}",
>  24                 deps : ['main'],
>  25                 config : {
>  26                     'org/forgerock/commons/ui/common/main/i18nManager': {
>  27                         i18nLoad: 'none'
>  28                     }
>  29                 }
>  30             };
>  31         </script>
> ```
> In this case, it works even without `ja-JP`.


Thank you for giving me a simple and good proposal.

The argument to 'i18nLoad' will work as expected even though it is 'none', but I think that 'all' is better in accordance with the current API specification.
https://www.i18next.com/overview/configuration-options#languages-namespaces-resources

Checking the version 1.x source code seems to be OK if it is other than 'unspecific' and 'current'.
https://github.com/i18next/i18next/blob/1.11.x/i18next.js#L856-L857

is that okay?

> i18nLoad: 'all'

**Comment by tsujiguchitky** [comment]
> i18nLoad: 'all'

Good point! I totally agree with you.

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json (hunk 19 above)
Comment (tsujiguchitky) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: That makes sense.

**Comment by tsujiguchitky** [comment]
By the way, is device.json covered by this issue?

**Comment by tsujiguchitky** [comment]
device.json is userd for `OAuth 2.0 Device Flow`.
This file translates the user_code input screen and completion screen.

**Comment by ogis-miyamura** [comment]
> device.json is userd for `OAuth 2.0 Device Flow`.
> This file translates the user_code input screen and completion screen.

Here it is.
Thank you for the information.

File: openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json [hunk 20]
```diff
@@ -0,0 +1,1291 @@
...
+                    "addClassNameDialog": {
+                        "title": "新しいクラス名の追加",
+                        "message": "ポスト認証プロセスの新規クラス <mark>__newClassName__</mark> は、保存されていません。保存しますか?"
+                    },
+                    "duplicateClass" : "同じ名前のクラスがすでに存在します",
+                    "help" : "認証プロセス終了時に実行する Java クラスの名前を指定します。"
+                },
+                "criteria" : [
+                    {
+                        "title": "必須",
```
[the remark this answers is not in the document]
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: Certainly I was often confused with these selections.
But It is easy to understand with the arrow in GUI, and there are hint balloons, so we will not worry about mistaking the operation.

If I express their behavior, my proposal is as follows:

- authentication.config.required.label  -> 必須(失敗時継続)
- authentication.config.requisite.label -> 必須(失敗時中断)

Do you have any idea?

[the remark this answers is not in the document]
Comment (tsujiguchitky) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I think that the proposal is easier to understand than the current translation. If you commit this content, I will approve it.

However, personally, these words are technical terms, so I think that it is better to leave these words in English.

[the remark this answers is not in the document]
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I agree that you pointed out that it is a technical term.

However I think it is not easy to choose the right one because:
 - People of various roles (including end users, SI team) are use this UI.
 - When multiple languages mixed in the dropdown list, Users may think that it is strange or uncomfortable.

So, I will ask my colleagues for opinions.
Please give me little more time.

[the remark this answers is not in the document]
Comment (tsujiguchitky) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: > When multiple languages mixed in the dropdown list, Users may think that it is strange or uncomfortable.

I'm sorry I made you misunderstand.
I think that it is better to leave `Optional` and `Sufficient` in English as well.
In that case, multiple languages will not be mixed in the dropdown list.

> So, I will ask my colleagues for opinions.

I appreciate it.

[the remark this answers is not in the document]
Comment (ogis-miyamura) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: I'm so sorry for my late reply.

I agree with the policy to leave criteria items in English.
My colleague's opinion was that it is difficult to choose the right item by looking at the item name in any Japanese translation.

[the remark this answers is not in the document]
Comment (tsujiguchitky) [reply openam-ui/openam-ui-ria/src/main/resources/locales/ja/translation.json]: Thank you for your consideration.

> it is difficult to choose the right item by looking at the item name in any Japanese translation.

That's true.
